### PR TITLE
MOVE-848: Fiksa henting av data frå enhetsregisteret slik at dersom e…

### PR DESCRIPTION
**Pull request by livarb**
…in enhet ikkje har postadresse, så blir forretningsadresse lagt inn som postadresse i staden for.

### CODE REVIEW

File: serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/model/OrganizationInfo.java [hunk 1]
```diff
@@ -1,9 +1,12 @@
 package no.difi.meldingsutveksling.serviceregistry.model;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.base.Strings;
```
Comment (nuth) [inline serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/model/OrganizationInfo.java]: MINOR Remove this unused import 'com.google.common.base.Strings'. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by nuth** [comment]
SonarQube analysis reported 22 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-websocket-8.5.31.jar | Reference: CVE-2016-6325 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 5 through 7, JBoss Web Server 3.0, and JBoss EWS 2 uses weak permissions for (1) /etc/sysconfig/tomcat and (2) /etc/tomcat/tomcat.conf, which allows local users to gain privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-websocket-8.5.31.jar | Reference: CVE-2016-5425 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 7, Fedora, CentOS, Oracle Linux, and possibly other Linux distributions uses weak permissions for /usr/lib/tmpfiles.d/tomcat.conf, which allows local users to gain root privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2016-6325 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 5 through 7, JBoss Web Server 3.0, and JBoss EWS 2 uses weak permissions for (1) /etc/sysconfig/tomcat and (2) /etc/tomcat/tomcat.conf, which allows local users to gain privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2016-5425 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 7, Fedora, CentOS, Oracle Linux, and possibly other Linux distributions uses weak permissions for /usr/lib/tmpfiles.d/tomcat.conf, which allows local users to gain root privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-2185 | CVSS Score: 7.5 | Category: CWE-20 Improper Input Validation | ** DISPUTED ** The readObject method in the DiskFileItem class in Apache Tomcat and JBoss Web, as used in Red Hat JBoss Enterprise Application Platform 6.1.0 and Red Hat JBoss Portal 6.0.0, allows remote attackers to write to arbitrary files via a NULL byte in a file name in a serialized instance, a similar issue to CVE-2013-2186.  NOTE: this issue is reportedly disputed by the Apache Tomcat team, although Red Hat considers it a vulnerability. The dispute appears to regard whether it is the responsibility of applications to avoid providing untrusted data to be deserialized, or whether this class should inherently protect against this issue. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![CRITICAL][CRITICAL] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2009-3548 | CVSS Score: 7.5 | Category: CWE-255 Credentials Management | The Windows installer for Apache Tomcat 6.0.0 through 6.0.20, 5.5.0 through 5.5.28, and possibly earlier versions uses a blank default password for the administrative user, which allows remote attackers to gain privileges. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-websocket-8.5.31.jar | Reference: CVE-2017-6056 | CVSS Score: 5.0 | Category: CWE-19 Data Handling | It was discovered that a programming error in the processing of HTTPS requests in the Apache Tomcat servlet and JSP engine may result in denial of service via an infinite loop. The denial of service is easily achievable as a consequence of backporting a CVE-2016-6816 fix but not backporting the fix for Tomcat bug 57544. Distributions affected by this backporting issue include Debian (before 7.0.56-3+deb8u8 and 8.0.14-1+deb8u7 in jessie) and Ubuntu. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2017-6056 | CVSS Score: 5.0 | Category: CWE-19 Data Handling | It was discovered that a programming error in the processing of HTTPS requests in the Apache Tomcat servlet and JSP engine may result in denial of service via an infinite loop. The denial of service is easily achievable as a consequence of backporting a CVE-2016-6816 fix but not backporting the fix for Tomcat bug 57544. Distributions affected by this backporting issue include Debian (before 7.0.56-3+deb8u8 and 8.0.14-1+deb8u7 in jessie) and Ubuntu. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0119 | CVSS Score: 4.3 | Category: CWE-264 Permissions, Privileges, and Access Controls | Apache Tomcat before 6.0.40, 7.x before 7.0.54, and 8.x before 8.0.6 does not properly constrain the class loader that accesses the XML parser used with an XSLT stylesheet, which allows remote attackers to (1) read arbitrary files via a crafted web application that provides an XML external entity declaration in conjunction with an entity reference, related to an XML External Entity (XXE) issue, or (2) read files associated with different web applications on a single Tomcat instance via a crafted web application. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0099 | CVSS Score: 4.3 | Category: CWE-189 Numeric Errors | Integer overflow in java/org/apache/tomcat/util/buf/Ascii.java in Apache Tomcat before 6.0.40, 7.x before 7.0.53, and 8.x before 8.0.4, when operated behind a reverse proxy, allows remote attackers to conduct HTTP request smuggling attacks via a crafted Content-Length HTTP header. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by martinwam** [comment]
build

**Comment by martinwam** [comment]
build

**Comment by nuth** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 12 major
* ![MINOR][MINOR] 1 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0119 | CVSS Score: 4.3 | Category: CWE-264 Permissions, Privileges, and Access Controls | Apache Tomcat before 6.0.40, 7.x before 7.0.54, and 8.x before 8.0.6 does not properly constrain the class loader that accesses the XML parser used with an XSLT stylesheet, which allows remote attackers to (1) read arbitrary files via a crafted web application that provides an XML external entity declaration in conjunction with an entity reference, related to an XML External Entity (XXE) issue, or (2) read files associated with different web applications on a single Tomcat instance via a crafted web application. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0099 | CVSS Score: 4.3 | Category: CWE-189 Numeric Errors | Integer overflow in java/org/apache/tomcat/util/buf/Ascii.java in Apache Tomcat before 6.0.40, 7.x before 7.0.53, and 8.x before 8.0.4, when operated behind a reverse proxy, allows remote attackers to conduct HTTP request smuggling attacks via a crafted Content-Length HTTP header. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0096 | CVSS Score: 4.3 | Category: CWE-264 Permissions, Privileges, and Access Controls | java/org/apache/catalina/servlets/DefaultServlet.java in the default servlet in Apache Tomcat before 6.0.40, 7.x before 7.0.53, and 8.x before 8.0.4 does not properly restrict XSLT stylesheets, which allows remote attackers to bypass security-manager restrictions and read arbitrary files via a crafted web application that provides an XML external entity declaration in conjunction with an entity reference, related to an XML External Entity (XXE) issue. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2014-0075 | CVSS Score: 5.0 | Category: CWE-189 Numeric Errors | Integer overflow in the parseChunkHeader function in java/org/apache/coyote/http11/filters/ChunkedInputFilter.java in Apache Tomcat before 6.0.40, 7.x before 7.0.53, and 8.x before 8.0.4 allows remote attackers to cause a denial of service (resource consumption) via a malformed chunk size in chunked transfer coding of a request during the streaming of data. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-6357 | CVSS Score: 6.8 | Category: CWE-352 | ** DISPUTED ** Cross-site request forgery (CSRF) vulnerability in the Manager application in Apache Tomcat 5.5.25 and earlier allows remote attackers to hijack the authentication of administrators for requests that manipulate application deployment via the POST method, as demonstrated by a /manager/html/undeploy?path= URI.  NOTE: the vendor disputes the significance of this report, stating that "the Apache Tomcat Security team has not accepted any reports of CSRF attacks against the Manager application ... as they require a reckless system administrator." [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-4590 | CVSS Score: 4.3 | Category: CWE-200 Information Exposure | Apache Tomcat before 6.0.39, 7.x before 7.0.50, and 8.x before 8.0.0-RC10 allows attackers to obtain "Tomcat internals" information by leveraging the presence of an untrusted web application with a context.xml, web.xml, *.jspx, *.tagx, or *.tld XML document containing an external entity declaration in conjunction with an entity reference, related to an XML External Entity (XXE) issue. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-4444 | CVSS Score: 6.8 | Category: CWE-94 Improper Control of Generation of Code ('Code Injection') | Unrestricted file upload vulnerability in Apache Tomcat 7.x before 7.0.40, in certain situations involving outdated java.io.File code and a custom JMX configuration, allows remote attackers to execute arbitrary code by uploading and accessing a JSP file. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-4322 | CVSS Score: 4.3 | Category: CWE-20 Improper Input Validation | Apache Tomcat before 6.0.39, 7.x before 7.0.50, and 8.x before 8.0.0-RC10 processes chunked transfer coding without properly handling (1) a large total amount of chunked data or (2) whitespace characters in an HTTP header value within a trailer field, which allows remote attackers to cause a denial of service by streaming data.  NOTE: this vulnerability exists because of an incomplete fix for CVE-2012-3544. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2013-4286 | CVSS Score: 5.8 | Category: CWE-20 Improper Input Validation | Apache Tomcat before 6.0.39, 7.x before 7.0.47, and 8.x before 8.0.0-RC3, when an HTTP connector or AJP connector is used, does not properly handle certain inconsistent HTTP request headers, which allows remote attackers to trigger incorrect identification of a request's length and conduct request-smuggling attacks via (1) multiple Content-Length headers or (2) a Content-Length header and a "Transfer-Encoding: chunked" header.  NOTE: this vulnerability exists because of an incomplete fix for CVE-2005-2090. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. ![MAJOR][MAJOR] no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-annotations-api-8.5.31.jar | Reference: CVE-2012-5568 | CVSS Score: 5.0 | Category: CWE-16 Configuration | Apache Tomcat through 7.0.x allows remote attackers to cause a denial of service (daemon outage) via partial HTTP requests, as demonstrated by Slowloris. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by martinwam** [comment]
merge

**Pull request closed without merging**